File: java/Disk.java
/*
    Saya Mohammad Ray Mosaid mengerjakan Latihan 3 dalam mata kuliah DPBO
    untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan.
    Aamiin.
*/

//membuat class Disk
class Disk {
    //membuat variabel class disk
    private String type;
    private int capacity;
    private int price;

    //membuat konstruktor
    public Disk(String type, int capacity, int price){
        this.type = type;
        this.capacity = capacity;
        this.price = price;
    }

    //membuat konstruktor dan juga prosedur set
    public Disk(){
    }
    public void settype(String type){
        this.type = type;
    }
    public void setcapacity(int capacity){
        this.capacity = capacity;
    }
    public void setprice(int price){
        this.price = price;
    }
    
    //membuat fungsi get untuk me-return atribut private
    public String gettype(){
        return type;
    }
    public int getcapacity(){
        return capacity;
    }
    public int getdiskprice(){
        return price;
    }

    //membuat prosedur print data disk
    public void printdiskdata(){
        System.out.println("Disk:");
        System.out.println("    Disk type: " + type);
        System.out.println("    Capacity: " + capacity + "TB");
        System.out.println("    Price: $" + price);
    }
}

File: c++/Disk.cpp
/*
    Saya Mohammad Ray Mosaid mengerjakan Latihan 3 dalam mata kuliah DPBO
    untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan.
    Aamiin.
*/

//memanggil header yang dibutuhkan
#include <iostream>
#include <string.h>

using namespace std;

//membuat class Disk
class Disk{
    //membuat variabel class disk
    private:
        string type;
        int capacity;
        int price;
    
    //membuat konstruktor
    public:
        //membuat konstruktor
        Disk(string type, int capacity, int price){
            this->type = type;
            this->capacity = capacity;
            this->price = price;
        }

        //membuat konstruktor dan juga prosedur set
        Disk(){
        }
        void settype(string type){
            this->type = type;
        }
        void setcapacity(int capacity){
            this->capacity = capacity;
        }
        void setdiskprice(int price){
            this->price = price;
        }

        //membuat fungsi get untuk me-return atribut private
        string gettype(){
            return this->type;
        }
        int getcapacity(){
            return this->capacity;
        }
        int getdiskprice(){
            return this->price;
        }

        //membuat prosedur print data disk
        void printdiskdata(){
            cout << "Disk:" << endl;
            cout << "   Disk Type: " << this->type << endl;
            cout << "   Capacity: " << this->capacity << "TB" << endl;
            cout << "   Price: $" << this->price << endl;
        }

        //membuat destruktor
        ~Disk(){
        }
};
